op-chain-ops/deployer/deployer.go

```diff
@@ -148,7 +112,11 @@ func Deploy(backend *backends.SimulatedBackend, constructors []Constructor, cb D
 		// The simulator performs asynchronous processing,
 		// so we need to both commit the change here as
 		// well as wait for the transaction receipt.
+
+		// give transaction time to hit mempool otherwise we might commit an empty block
+		time.Sleep(100 * time.Millisecond)
```

Comment:
I know these sleep calls are a bit awkward and I worry a bit whether it will lead to (more) flakiness, but without them the tests fail reliably when run in parallel due to the new behavior of the simulated backend.    Open to other ideas around how to fix this.

Reply:
Rather than sleeping could we modify `backend.Commit` to a new function `backend.CommitWithTransaction` which would use the payload attributes `Transactions` field. That field is specific to Optimism, not upstream, but would really simplify this code.

Reply:
Just had a better idea (I think):  implemented a local version of WaitMined() which calls commit in a loop.

PTAL?

Reply:
Retry loop LGTM. I was initially worried that producing multiple block would be an issues, but upon closer inspection, that is not true & the retry loop will work.